File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}
        for i, number in enumerate(nums):
            if target - number in hm:
                return [ hm[target - number], i]
            hm[number] = i
            
        
            
            

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        int n = nums.length;
        int[] res = new int[2];
        
        for (int i = 0; i < n; i++) {
            hm.put(target - nums[i], i);
        }
        
        for (int i = 0; i < n; i++) {
            if (hm.containsKey(nums[i]) && hm.get(nums[i]) != i) {
                res[0] = i;
                res[1] = hm.get(nums[i]);
                return res;
            }
        }
        
        return res;
    }
}